CMakeLists.txt

```diff
@@ -101,24 +101,38 @@ if(UNIX AND NOT ES_USE_VCPKG)
 		add_library(FLAC::FLAC++ ALIAS PkgConfig::flac++)
 	endif()
 else()
-	find_package(FLAC CONFIG REQUIRED)
+	find_package(FLAC CONFIG QUIET)
+	if(NOT FLAC_FOUND AND APPLE)
+		# Homebrew's FLAC doesn't provide CMake config files, use PkgConfig instead.
+		find_package(PkgConfig REQUIRED)
+		pkg_check_modules(flac++ REQUIRED IMPORTED_TARGET flac++)
+		add_library(FLAC::FLAC++ ALIAS PkgConfig::flac++)
+	elseif(NOT FLAC_FOUND)
+		find_package(FLAC CONFIG REQUIRED)
+	endif()
 endif()
 
 # Find the zip-reading component of zlib.
 find_package(ZLIB REQUIRED)
 
 # Minizip has a different target name in vcpkg.
 if(ES_USE_VCPKG)
-	find_package(unofficial-minizip CONFIG REQUIRED)
-	set(MINIZIP_LIBRARIES "unofficial::minizip::minizip")
-else()
-	if(CMAKE_VERSION VERSION_LESS 3.31)
-		find_package(PkgConfig REQUIRED PkgConfig PkgConf)
-		pkg_check_modules(MINIZIP REQUIRED minizip)
+	find_package(unofficial-minizip CONFIG QUIET)
+	if(NOT unofficial-minizip_FOUND AND APPLE)
+		# Homebrew's minizip doesn't provide CMake config files, use PkgConfig instead.
+		find_package(PkgConfig REQUIRED)
+		pkg_check_modules(MINIZIP REQUIRED IMPORTED_TARGET minizip)
+		add_library(Minizip::Minizip ALIAS PkgConfig::MINIZIP)
+		set(MINIZIP_LIBRARIES "Minizip::Minizip")
 	else()
-		cmake_pkg_config(EXTRACT minizip REQUIRED)
-		set(MINIZIP_LIBRARIES "minizip")
+		find_package(unofficial-minizip CONFIG REQUIRED)
+		set(MINIZIP_LIBRARIES "unofficial::minizip::minizip")
 	endif()
+else()
+	find_package(PkgConfig REQUIRED)
+	pkg_check_modules(MINIZIP REQUIRED IMPORTED_TARGET minizip)
+	add_library(Minizip::Minizip ALIAS PkgConfig::MINIZIP)
+	set(MINIZIP_LIBRARIES "Minizip::Minizip")
```

Comment:
If this is the new default behavior when not using vcpkg, rewrite it not to introduce pkgconfig as a dependency.

Reply:
Alright, so should I go with something like this, or do you have a better alternative?
```
		find_path(FLAC_INCLUDE_DIR FLAC++/decoder.h
			PATHS /opt/homebrew/include /usr/local/include)
		find_library(FLAC_LIBRARY NAMES flac++ libflac++
			PATHS /opt/homebrew/lib /usr/local/lib)
		if(FLAC_INCLUDE_DIR AND FLAC_LIBRARY)
			add_library(FLAC::FLAC++ UNKNOWN IMPORTED)
			set_target_properties(FLAC::FLAC++ PROPERTIES
				IMPORTED_LOCATION "${FLAC_LIBRARY}"
				INTERFACE_INCLUDE_DIRECTORIES "${FLAC_INCLUDE_DIR}")
		else()
			message(FATAL_ERROR "FLAC++ not found. Please install flac via Homebrew")
		endif()
```

This feels wrong to me, but I'm not sure how else to accomplish this.